fw/src/mppt/global/trapeziummethod.hpp

```diff
@@ -0,0 +1,193 @@
+/**
+ * @file trapeziummethod.hpp
+ * @author Afnan Mir (afnanmir@utexas.edu) & Nash Wu
+ * @brief Implementation of Trapezoidal Sum Optimization GlobalMPPTAlgortihm.
+ * @version 0.2.0
+ * @date 2024-01-20
+ * 
+ * @copyright Copyright (c) 2024
+ * 
+ */
+#pragma once
+#include "../mppt.hpp"
+#include "../local/pando.hpp"
+
+
+#define DV 0.5
+
+class TrapeziumMethod final : public MPPT {
+    public:
+        TrapeziumMethod(void) : MPPT() {
+            reference_voltage = 0.0;
+            prev_array_voltage = 0.0;
+            prev_array_power = 0.0;
+            prev_array_current = 0.0;
+
+            prev_area = 0.0;
+
+            findingTrapezoids = true;
+            startLocal = true;
+            kick = true;
+
+        }
+
+        float getReferenceVoltage(void) {
+            float reference_voltage = array_voltage;
```

Comment:
Has someone done a simple sanity test on this to verify that it works?
Or at least provide a decision tree/state diagram so third parties can look at this and verify that the code matches the expected actions